web/src/main/java/com/bazaarvoice/emodb/web/auth/ApiKeyAdminTask.java

```diff
@@ -332,7 +352,8 @@ private void migrateApiKey(ImmutableMultimap<String, String> parameters, PrintWr
         output.println("\nWarning:  This is your only chance to see this key.  Save it somewhere now.");
     }
 
-    private void deleteApiKey(ImmutableMultimap<String, String> parameters, PrintWriter output) {
+    private void deleteApiKey(Subject subject, ImmutableMultimap<String, String> parameters, PrintWriter output) {
+        subject.checkPermission(Permissions.deleteApiKey());
         String key = getValueFromParams("key", parameters);
```

Comment:
That's a great question.  My reasoning is this:

1. The grant-role check when removing a role from an API key is to prevent an accidental or possibly malicious user from revoking permissions which he himself cannot grant back.  Basically, the grant-role permission is being used as a proxy for group ownership.
2. When an API key is being deleted there is no API key left for which the role can be granted back.  Therefore, rather than requiring permission to grant each individual role a much higher and more protected delete-api-key permission is needed.

It's possible that an API key can exist which has delete-api-key permission but not grant-role permission for all roles on an API key.  However, I don't see a use case for this since, unlike API key creation, deletion in practice is tightly controlled and limited only to Emo administrators.  It seems like a bad practice to delegate this particular permission to non-trusted administrators.

TL;DR:  I have no objection to adding this check, it just seemed redundant since any user with delete permission would be an administrator.